Packs/Uptycs/Integrations/Uptycs/Uptycs.py

```diff
@@ -367,12 +451,166 @@ def uptycs_get_assets_command():
+    if asset_type != -1 and asset_type is not None:
+        filters.append('"agentType":{"in":["%s"]}' % asset_type)
+
+    time_ago = demisto.args().get('time_ago')
+    start_window = demisto.args().get('start_window')
+    end_window = demisto.args().get('end_window')
+
+    if time_ago is not None or (start_window is not None or end_window is not None):
+        begin, end = uptycs_parse_date_range(time_ago, start_window, end_window)
+        filters.append('"createdAt":{"between":["%s","%s"]}' % (begin, end))
+
+    if len(filters) > 0:
+        query_parameters.append("filters={%s}" % ','.join(filters))
+
+    if len(query_parameters) > 0:
+        api_call = ("%s?%s" % (api_call, '&'.join(query_parameters)))
+
+    return restcall(http_method, api_call)
+
+
+def uptycs_get_detections_command():
+    query_results = uptycs_get_detections()
+    context = query_results.get('items')
+    context_entries_to_keep = ['id', 'displayName', 'severity', 'signals',
+                               'status', 'tacticCount', 'tactics',
+                               'assetHostName', 'assetId', 'agentType',
+                               'assetLive', 'assetLocation', 'assetOs',
+                               'assetOsFlavor', 'assetOsVersion', 'assetOsqueryVersion',
+                               'assetStatus', 'assignedTo', 'assignedUserName',
+                               'createdAt', 'lastOccurredAt']
+
+    if context is not None:
+        remove_context_entries(context, context_entries_to_keep)
+
+    human_readable = tableToMarkdown('Uptycs Detections: ',
+                                     context,
+                                     context_entries_to_keep)
+    entry = {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': query_results,
+        'HumanReadable': human_readable,
+        'EntryContext': {
+            'Uptycs.Detections(val.id == obj.id)': context
+        }
+    }
+
+    return entry
+
+
+def uptycs_get_detection_details():
+    """
+    return detection details
+    """
+    http_method = 'get'
+    api_call = "/detections/"
+    detection_id = demisto.args().get('detection_id')
+
+    if detection_id != -1 and detection_id is not None:
+        api_call = ("%s%s" % (api_call, detection_id))
+
+    return restcall(http_method, api_call)
+
+
+def uptycs_get_detection_details_command():
+    query_results = uptycs_get_detection_details()
+    detection = query_results.get('detection')
+    final_alerts = []
+    detection_url = 'https://' + DOMAIN + '/ui/detections/' + demisto.args().get('detection_id')
+    display_name = get_href_link(detection["displayName"], detection_url)
+
+    for alert in query_results.get('alerts').get('items'):
+        alert_text = alert['alertTime'] + '<br>Alert - ' + alert['alertRuleName']
+        alert_text += '<br>\tscore - ' + alert['score'] + '<br>\t'
+        alert_text += alert['key'] + ' : ' + alert['value'] + '<br>\t'
```

Comment:
```suggestion
        alert_text += alert.get('key', "") + ' : ' + alert.get('value', "") + '<br>\t'
```
Let's refrain from direct access to dict elements (for all other locations in the code).